src/sbvr-api/sbvr-utils.coffee

```diff
@@ -685,11 +685,15 @@ exports.handleODataRequest = handleODataRequest = (req, res, next) ->
 	mapSeries = controlFlow.getMappingFn(req.headers)
 	# Get the hooks for the current method/vocabulary as we know it,
 	# in order to run PREPARSE hooks, before parsing gets us more info
-	req.hooks = getHooks(
+	req.hooks = reqHooks = getHooks(
 		method: req.method
 		vocabulary: apiRoot
 	)
-	runHooks('PREPARSE', { req, tx: req.tx })
+
+	req.tx?.on 'rollback', ->
```

Comment:
This is only needed if the `tx` is created in a hook right? If we are in an internal request shouldn't the rollback handler be already setup from the runTransaction callback (lines 729-731)?

Reply:
The rollback handler for the top level request would be set up there, but internal requests have their own sets of hooks that need to be rolled back and this attaches their rolling back to happen as part of the parent's tx rolling back